### DISCUSSION TITLE
Playground seems broken after today's Pull

### DISCUSSION THREAD

**Discussion by cainezen** [oobabooga/text-generation-webui#5143] [General]
Anyone else see that the Playground extension broke after a pull today?  Yesterday it was working fine but when I just did a Pull, I'm getting errors. Just curious if this is a known issue.  Error: 
Closing server running on port: 7860
18:47:59-142431 INFO     Loading the extension "gallery"
18:47:59-144431 INFO     Loading the extension "Playground"
18:47:59-147432 ERROR    Failed to load the extension "Playground".
Traceback (most recent call last):
  File "E:\StableLLM\Dec23\text-generation-webui\modules\extensions.py", line 37, in load_extensions
    exec(f"import extensions.{name}.script")
  File "<string>", line 1, in <module>
  File "E:\StableLLM\Dec23\text-generation-webui\extensions\Playground\script.py", line 12, in <module>
    from modules.LoRA import add_lora_autogptq, add_lora_exllama, add_lora_exllamav2
ImportError: cannot import name 'add_lora_exllama' from 'modules.LoRA' (E:\StableLLM\Dec23\text-generation-webui\modules\LoRA.py)
--Update ---
I was able to at least suppress this error by removing the reference to "add_lora_exllama" in the script.py of Playground.

**Comment by TheLounger** [comment]
ExLlama (v1) was [removed](https://github.com/oobabooga/text-generation-webui/commit/0e54a09bcbca4bc69dcc1c887f92d3fefe4e5f6a) recently, so the extension would need to be updated.

**Reply by cainezen** [reply]
I figured it was some kind of deprecated feature.  I use Playground religiously, so hopefully they update that extension.

**Reply by DelteRT** [reply]
I removed the lines referring to ExLlama (v1) and it is working again, if someone needs a quick and temporary solution, the lines to modify are (In the main script.py file):

**Before:** `from modules.LoRA import add_lora_autogptq, add_lora_exllama, add_lora_exllamav2`
**After:** `from modules.LoRA import add_lora_autogptq, add_lora_exllamav2`

**Before:**

```
    def add_lora_to_model(lora_name):
        if 'GPTQForCausalLM' in shared.model.__class__.__name__ or shared.args.loader == 'AutoGPTQ':
            add_lora_autogptq([lora_name])
        elif shared.model.__class__.__name__ in ['ExllamaModel', 'ExllamaHF'] or shared.args.loader == 'ExLlama':
            add_lora_exllama([lora_name])
        elif shared.model.__class__.__name__ in ['Exllamav2Model', 'Exllamav2HF'] or shared.args.loader == ['ExLlamav2', 'ExLlamav2_HF']:
            add_lora_exllamav2([lora_name])
        else:
```

**After:**

```
    def add_lora_to_model(lora_name):
        if 'GPTQForCausalLM' in shared.model.__class__.__name__ or shared.args.loader == 'AutoGPTQ':
            add_lora_autogptq([lora_name])
        elif shared.model.__class__.__name__ in ['Exllamav2Model', 'Exllamav2HF'] or shared.args.loader == ['ExLlamav2', 'ExLlamav2_HF']:
            add_lora_exllamav2([lora_name])
        else:
```


_I'm not a programmer, and this will most likely break something related to the lora models, but if someone like me uses LLMs for work, and can't wait a couple of days, it's a temporary solution that seems to work_

**Reply by cainezen** [reply]
Looks like it's broken again, this time on "AutoGPTQ".  I had to remove the first two lines and just leave the "if" statement with ExLlamaV2.

**Comment by bcauldron** [comment]
Hello,
I have the same problem, I have updated the webui and Playground (which was my favorite extension for writing short stories) doesn't run anymore... I've tried your suggestions but it still doesn't work...

Is it still possible to use Playground? And how?...

And, if not, do you know of another similar extension?...

Thanks for your help.

**Reply by DelteRT** [reply]
Whenever I run into issues like this, I always end up with the same solution: just go back to the version where everything was working fine

I'm using a version from October last year. Everything runs perfectly, and I don't have to deal with things randomly breaking

Try going back to the previous version, wait a couple of updates and try again

**Reply by bcauldron** [reply]
Thanks for the suggestion.
After several attempts, I managed to get the program working again. It's a pity though, I hope it will be updated but the developer hasn't touched it for two years so I'm moderately optimistic...
